### PR TITLE
Popup to create handler

### PR DESCRIPTION
**Pull request by ustundag**
show popup to enable users to easily create handler classes. 
adding "browse panel" feature may be better to choose file location.

### CODE REVIEW

**Comment by ustundag** [comment]
Example 

createNewHandlerDemo

File: src/main/kotlin/org/bilalkilic/kediatrhelper/listeners/KediatrHandlerClassMarker.kt [hunk 1]
```diff
@@ -51,9 +63,41 @@ class KediatrHandlerClassMarker : LineMarkerProvider {
...
+            .setResizable(false)
+            .setRequestFocus(true)
+            .setCancelOnWindowDeactivation(false)
+            .setItemChosenCallback { element.project.service<CreateNewHandlerService>().create(element) }
+            .createPopup()
+        popup.show(RelativePoint(Point(me.xOnScreen + 30, me.yOnScreen + 20)))
+    }
+
+    private fun showPopupWithIcon(element: PsiElement, me: MouseEvent) {
+        val mainClass = (element.parent as KtFile).classes.firstOrNull() ?: return
```
Comment (bilal-kilic) [inline src/main/kotlin/org/bilalkilic/kediatrhelper/listeners/KediatrHandlerClassMarker.kt]: This doesn't work if the file contains more than one classes ie:

```
class ExistingCommand: Command

class ClassIWantToCreateHandlerFor : Command
```
it tries to create a handler for the existing command. Instead, we could use KtClass which is already filtered in getLineMarkerInfo. Also, I'm not sure this casting is safe `(element.parent as KtFile)`  🤔

File: src/main/kotlin/org/bilalkilic/kediatrhelper/listeners/KediatrHandlerClassMarker.kt [hunk 2]
```diff
@@ -51,9 +63,41 @@ class KediatrHandlerClassMarker : LineMarkerProvider {
...
+        val title = POPUP_TITLE_PREFIX + mainClass.name
+        val options = listOf(POPUP_STEP_MESSAGE)
+        val step = object : BaseListPopupStep<String>(title, options, Icons.createNewHandlerGutter) {
+            override fun onChosen(selectedValue: String, finalChoice: Boolean): PopupStep<*>? {
```
Comment (bilal-kilic) [inline src/main/kotlin/org/bilalkilic/kediatrhelper/listeners/KediatrHandlerClassMarker.kt]: Maybe we could add creating async handlers as an option too.

File: src/main/kotlin/org/bilalkilic/kediatrhelper/services/CreateNewHandlerService.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import org.bilalkilic.kediatrhelper.utils.getQueryReturnType
+import org.bilalkilic.kediatrhelper.utils.isQuery
+import org.jetbrains.kotlin.psi.KtFile
+
+@Service
+class CreateNewHandlerService {
+
+    fun create(element: PsiElement) {
+        val directory = element.containingFile.containingDirectory
+        val mainClass = (element.parent as KtFile).classes.firstOrNull() ?: return
```
Comment (bilal-kilic) [inline src/main/kotlin/org/bilalkilic/kediatrhelper/services/CreateNewHandlerService.kt]: Again can change the element to KtClass to avoid casting here.

**Review by bilal-kilic — CHANGES_REQUESTED** [review]
Thanks for your contribution 🙏🏻. I commented on some issues.

**Comment by ustundag** [comment]
hi, thanks for your comments, that's a good point. I fixed bugs and applied new additions. it may still need your review though.

kediatr-plugin-async-handler

File: src/main/kotlin/org/bilalkilic/kediatrhelper/listeners/KediatrHandlerClassMarker.kt (hunk 1 above)
Comment (ustundag) [reply src/main/kotlin/org/bilalkilic/kediatrhelper/listeners/KediatrHandlerClassMarker.kt]: done.

File: src/main/kotlin/org/bilalkilic/kediatrhelper/services/CreateNewHandlerService.kt (hunk 3 above)
Comment (ustundag) [reply src/main/kotlin/org/bilalkilic/kediatrhelper/services/CreateNewHandlerService.kt]: done.

File: src/main/kotlin/org/bilalkilic/kediatrhelper/listeners/KediatrHandlerClassMarker.kt (hunk 2 above)
Comment (ustundag) [reply src/main/kotlin/org/bilalkilic/kediatrhelper/listeners/KediatrHandlerClassMarker.kt]: added.

**Review by bilal-kilic — APPROVED** [review]
I will be releasing these changes in upcoming version

**Pull request merged**